### PR TITLE
 👽 build compile_commands after project creation

### PR DESCRIPTION
**Pull request by HotelCalifornia**
### Summary

Should create the `compile_commands.json` file immediately after project creation, so cquery doesn't mistakenly tell users that their new project is full of errors

### Test Plan

- [x] project created successfully
- [x] project compiles after creation
- [x] `compile_commands.json` generated properly (no more cquery errors shown for fresh project)

#### References
T571

### CODE REVIEW

**Comment by HotelCalifornia** [comment]
note that two of the commits are already on master, not sure why or how they got duplicated here

**Comment by HotelCalifornia** [comment]
also note that this doesn't currently work as intended... debugging

File: lib/views/NewProjectView.js [hunk 1]
```diff
@@ -8,28 +8,36 @@ import { createNewProject } from '@purduesigbots/pros-cli-middleware'
...
+          glob(`${loc}/src/opcontrol.*`, (err, files) => {
             if(files.length > 0) {
               atom.workspace.open(files[0], {pending: true})
             }
           })
-        }}), // end of MiddlewareCallbackHelper
-        cmp.path, cmp.kernel
-      )
+        } else {
+          console.error(`build-compile-commands exited with error code ${code}`)
```
Comment (edjubuh) [inline lib/views/NewProjectView.js]: That's not the only reason why the return code could be 0... If we couldn't create the project for some reason.

**Review by edjubuh — DISMISSED** [review]
Other than the one small comment, looks good. Will approve after removing/changing the console statement

File: lib/views/NewProjectView.js (hunk 1 above)
Comment (HotelCalifornia) [reply lib/views/NewProjectView.js]: that's fair, but how will we be able to differentiate? can we assign different exit codes for the project creation and the build?

**Review by edjubuh — APPROVED** [review]

**Pull request merged**